compiler+runtime/src/cpp/jank/read/lex.cpp

```diff
@@ -260,11 +267,51 @@ namespace jank::read
       return none;
     }
 
-    static native_bool is_symbol_char(char const c)
+    static result<codepoint, error> convert_to_codepoint(native_persistent_string_view sv, size_t pos)
+    {
+      std::mbstate_t state{};
+      wchar_t wc{};
+      size_t len = std::mbrtowc(&wc, sv.data(), sv.size(), &state);
```

Comment:
Please use direct initialization and avoid spelling out types.

`auto const len{ std::mbrtowc(&wc, sv.data(), sv.size(), &state) }`;

Reply:
Just curious why not spell out types? I am not disagreeing, just want to understand the rationale.

Reply:
The compiler knows the type. We can find the type by looking at the expression. It's only due to historical reasons that the left hand side needed the type there at all.

Finally, it violates DRY by focusing on nominal typing. It means if we update the return type, even to something structurally compatible, we'll break compilation at every call site. Better to use `auto`.